### PR TITLE
Rubocop

### PR DESCRIPTION
**Pull request by RomainEndelin**
Setup Rubocop for code linting in Ruby.

## Description

This PR comes from a discussion with @kennethc about code style in Ruby. This codebase has many contributors from different background, and Rubocop will probably help with code consistency, as well as enforcing a few good practices.

Rubocop includes a set of rules (named cops), and its configuration can be fine-tuned. I tried to stick to the default as much as possible, with a few tweaks. The goal is to find a good balance between useful and pedantic.

## References
N/A

## Implementation notes

- I've sticked to the default, but decided to override a few cops in `.rubocop.yml`, each of them comes with reference to the cop documentation, and discussion on why it's overriden
- It would be tempting to use [Rubocop-Airbnb](https://github.com/airbnb/ruby/tree/master/rubocop-airbnb) for even better rules. But I'm afraid it's too much at once. If we do it, I'd rather open a new PR
- A few items are left in `.rubocop-todo.yml`, because they're a little harder to fix (or they may conflict with @kennethc upcoming bugfix)

## Tasks

Here's a small description of my workflow:

- [x] Install [Rubocop](https://github.com/rubocop-hq/rubocop) & [Rubocop-Rails](https://github.com/rubocop-hq/rubocop-rails)
- [x] First step: Generate a `.rubocop-todo.yml` file to ignore all existing rule infractions (while applying rules on newer code)
  - `bundle exec rubocop --auto-gen-config`
- [x] Next step: Remove lines from `.rubocop-todo.yml`, and auto-correct as many infractions as possible. Refine rules in `.rubocop.yml` when needed.
  - `bundle exec rubocop --auto-correct`
- [x] Final step: Manually fix remaining infractions in `.rubocop-todo.yml`

In parallel:
- [x] Add Rubocop to CI
- [x] Add Docs

## Risks (if any)
* Medium
Large code change, but supposedly harmless.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -34,6 +34,33 @@ jobs:
       - store_test_results:
           path: /tmp/test_reports
 
+  rubocop:
+    docker:
+      - image: circleci/ruby:2.5.1-node-browsers
```
Comment (xuebingli) [inline .circleci/config.yml]:
```suggestion
      - image: circleci/ruby:2.6.1-node-browsers
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -34,6 +34,33 @@ jobs:
...
+        environment:
+          RAILS_ENV: development
+          TESTOPTS: "--ci-report --ci-dir=/tmp/rubocop_reports/"
+      - image: circleci/postgres:10.3
```
Comment (xuebingli) [inline .circleci/config.yml]: I don't think we need database to run rubocop.

**Review by xuebingli — APPROVED** [review]
LGTM 👍 
What do you think @craig-day?

**Review by craig-day — APPROVED** [review]
Seems good enough for me! I don't love the rubocop defaults on some things, but it's a reasonable place to start :D

**Pull request merged**